spyder_kernels/customize/spyderpdb.py

```diff
@@ -214,11 +204,44 @@ def is_name_or_composed(text):
             text = text[1:]
             begidx += 1
 
-        matches = compfunc(text, line, begidx, endidx)
+        matches = []
+        if compfunc:
+            matches = compfunc(text, line, begidx, endidx)
+
+        cursor_start = cursor_pos - len(text)
+
+        if ipython_do_complete:
+            kernel = get_ipython().kernel
+            # Make complete call with current frame
+            if self.curframe:
+                kernel.shell.set_completer_frame(self.curframe)
+            result = super(SpyderKernel, kernel).do_complete(code, cursor_pos)
+            # Reset frame
+            kernel.shell.set_completer_frame()
+            # If there is no pdb results to merge, return the result
+            if not compfunc:
+                return result
+
+            ipy_matches = result['matches']
+            # Make sure both match lists start at the same place
+            if cursor_start < result['cursor_start']:
+                # Fill ipython matches
+                missing_txt = code[cursor_start:result['cursor_start']]
+                ipy_matches = [missing_txt + m for m in ipy_matches]
+            elif result['cursor_start'] < cursor_start:
+                # Fill pdb matches
+                missing_txt = code[result['cursor_start']:cursor_start]
+                matches = [missing_txt + m for m in matches]
+                cursor_start = result['cursor_start']
+
+            # Add pdb-specific matches
+            matches = matches + \
```

Comment:
I really don't like how this looks. Please remove the continuation line at the end and leave everything in a single line, even if it's wider than 79 characters. Or use an intermediate variable to save the result of

    [match for match in ipy_matches if match not in matches]